### ISSUE TITLE
Ability to reconcile row-level data based on hashed output

### ISSUE THREAD

**Issue by chadlwilson** [ThoughtWorks-SEA/recce#10]
**Context / Goal**
After #18 we have the ability to persist source and hashed data into the Recce DB. After that we can report on it to provide feedback on mismatched hashes. This should give a summary to the end user for review, and be able to extended later to provide more access to detailed results.

**Expected Outcome**

- Be able to compute and output
- For source and target individually
  - Number of rows loaded
  - Column names (and types?) in dataset
  - Metadata about the dataset? (Db version? driver version?)
- Combined results
  - \# matching rows
  - \# rows only in source (and migration keys or first 'n'?)
  - \# rows only in target (and migration keys or first 'n'?)
  - \# mismatched rows (and migration keys or first 'n'?)
  - %ages for the above

**Out of Scope**

- Don't worry about persisting summarised results or returning via any API initially, just compute them and get the model right

**Additional context / implementation notes**
